### DISCUSSION TITLE
Why is msys/mingw used?

### DISCUSSION THREAD

**Discussion by supsm** [microsoft/vcpkg#34522] [Q&A] [answered]
I am using vcpkg to install some packages on windows. I've noticed that there were many lines about using msys or mingw, for example
```
-- Using cached mingw-w64-i686-pkgconf-1~1.8.0-2-any.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.4.6-1-x86_64.pkg.tar.zst.
```
Visual studio is installed and the required environment variables are set, so how come it is still installing and using mingw/msys2 instead of msvc? I am mostly worried about being able to use these libraries from visual studio/msvc.
Target triplet is `x64-windows`, a package that has this is `brotli`.

**Comment by dg0yt** [comment, marked as answer]
It is about particular tools (and their runtime), not about (link) libraries.

The packages you list are for the `pkgconf` tool, an implementation of `pkg-config`. This is used to exchange usage information between builds of different packages. It supports MSVC syntax on Windows.

Other packages provide common Unix tools expected by the build system of many packages. 

For completeness it is "msys2" which is based on "mingw-w64".

**Reply by supsm** [reply]
That makes sense, thanks. I've noticed that the icu package has significantly more of these, including make, zlib, gcc libs, etc. Do these also serve the same purpose?

**Reply by dg0yt** [reply]
Yes. That's the broader standard list of tools which generally can be expected on Unix systems but are missing (at least in a compatible implementation) on Windows.
